packages/firestore/src/remote/datastore.ts

```diff
@@ -232,6 +237,29 @@ export async function invokeRunQueryRpc(
   );
 }
 
+export async function invokeRunAggregationQueryRpc(
+  datastore: Datastore,
+  aggregateQuery: AggregateQuery
+): Promise<ProtoValue[]> {
+  const datastoreImpl = debugCast(datastore, DatastoreImpl);
+  const request = toRunAggregationQueryRequest(
+    datastoreImpl.serializer,
+    queryToTarget(aggregateQuery.query._query)
+  );
+  const response = await datastoreImpl.invokeStreamingRPC<
+    ProtoRunAggregationQueryRequest,
+    ProtoRunAggregationQueryResponse
+  >('RunAggregationQuery', request.parent!, {
+    structuredAggregationQuery: request.structuredAggregationQuery
+  });
+  return (
+    response
+      // Omit RunQueryResponses that only contain readTimes.
+      .filter(proto => !!proto.result && !!proto.result.aggregateFields)
```

Comment:
Did you find that you need to check `!!proto.result.aggregateFields`? If it is not strictly necessary, please remove it.

Reply:
Without checking `!!proto.result.aggregateFields`, the return type would be` (Value | undefined)[]`, and I need to map through the returned value in later process to filter `undefined` in the returned array.

Currently, I filtered out the ones doesn't have aggregateFields, so the returned value is either `value[]` or an empty array, no need to worry about undefined in later process.